monkey/infection_monkey/monkey.py

```diff
@@ -118,16 +118,16 @@ def __init__(self, args):
 
         self._agent_event_serializer_registry = self._setup_agent_event_serializers()
 
-        server, self._island_api_client = self._connect_to_island_api()
-        self._cmd_island_ip = server.ip
-        self._cmd_island_port = server.port
-
-        self._island_address = SocketAddress(self._cmd_island_ip, self._cmd_island_port)
+        self._island_address, self._island_api_client = self._connect_to_island_api()
+        self._cmd_island_ip = self._island_address.ip
+        self._cmd_island_port = self._island_address.port
 
         self._control_client = ControlClient(
-            server_address=str(server), island_api_client=self._island_api_client
+            server_address=str(self._island_address), island_api_client=self._island_api_client
+        )
+        self._control_channel = ControlChannel(
+            str(self._island_address), get_agent_id(), self._island_api_client
```

Comment:
What are these changes?

Reply:
It looks like we're holding onto `_island_address`. However, I don't see why that is warranted, since it is only used in this constructor

Reply:
Dropped the commit :D